### ISSUE TITLE
Difference behavior for `(` and `)` on windows and linux

### ISSUE THREAD

**Issue by evilebottnawi** [mrmlnc/fast-glob#160]
### Environment

  * OS Version: windows/linux
  * Node.js Version: 10.15.0 (no matter)

### Actual behavior

We have `[special?directory]/(special-*file).txt` (yep it is valid name on windows)

Pattern - `(special-*file).txt`
cwd - `[special?directory]`

Linux : `['(special-*file).txt']`
Windows: `[]`

### Expected behavior

Equals behavior.

### Steps to reproduce

Below

### Code sample

```js
fg(['(special-*file).txt'], { cwd: '[special?directory]' }).then((entries) => console.log(entries));
```

From CI https://ci.appveyor.com/project/webpack-contrib/copy-webpack-plugin/builds/22485753/job/rqpb76mp3wxtlfqh

**Comment by mrmlnc** [comment]
Hmm…, I see that on my Windows 10 machine (NTFS, File Explorer, Powershell) I cannot create directory with `?` symbol in the name:

https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file#naming-conventions

Any way this is a issue for micromatch or picomatch:

```js
// micromatch
/^(?:(?:\.[\\\/](?=.))?(?:special-[^\\\/]*?file)\.txt(?:[\\\/]|$))$/
// false

// minimatch
/^(?:(?=.)\(special-[^\/]*?file\)\.txt)$/
// true

// picomatch
/^(?:(?=.)(special\-[^\\\/]*?file)\.txt)$/
// false
```

I'll create a separate issue for picomatch.

**Comment by mrmlnc** [comment]
As we found out in the linked issue, this is normal behavior. So, I will describe this behavior in the `README.md`. I also plan to add a function that escapes all special characters in the pattern.

**Comment by mrmlnc** [comment]
This issue will be fixed with release `fast-glob@2.2.7`. You can check release issue #180.
